Flagsmith.FlagsmithClient/FlagsmithClient.cs

```diff
@@ -28,217 +28,190 @@ namespace Flagsmith
-        private int? Retries { get; set; }
-        private Dictionary<string, string> CustomHeaders { get; set; }
-        const string _defaultApiUrl = "https://edge.api.flagsmith.com/api/v1/";
-        private HttpClient httpClient;
-        private EnvironmentModel Environment { get; set; }
-        private readonly PollingManager _PollingManager;
-        private readonly IEngine _Engine;
-        private readonly AnalyticsProcessor _AnalyticsProcessor;
+        private readonly ILogger _logger;
+        private readonly IFlagsmithClientConfig _config;
+        private readonly HttpClient _httpClient;
+        private readonly PollingManager _pollingManager;
+        private readonly IEngine _engine;
+        private readonly AnalyticsProcessor _analyticsProcessor;
+        private EnvironmentModel _environment;
+
         /// <summary>
         /// Create flagsmith client.
         /// </summary>
-        /// <param name="environmentKey">The environment key obtained from Flagsmith interface.</param>
-        /// <param name="apiUrl">Override the URL of the Flagsmith API to communicate with.</param>
-        /// <param name="logger">Provide logger for logging polling info & errors which is only applicable when client side evalution is enabled and analytics errors.</param>
-        /// <param name="defaultFlagHandler">Callable which will be used in the case where flags cannot be retrieved from the API or a non existent feature is requested.</param>
-        /// <param name="enableAnalytics">if enabled, sends additional requests to the Flagsmith API to power flag analytics charts.</param>
-        /// <param name="enableClientSideEvaluation">If using local evaluation, specify the interval period between refreshes of local environment data.</param>
-        /// <param name="environmentRefreshIntervalSeconds"></param>
-        /// <param name="useLegacyIdentities">Enables local evaluation of flags.</param>
-        /// <param name="customHeaders">Additional headers to add to requests made to the Flagsmith API</param>
-        /// <param name="retries">Total http retries for every failing request before throwing the final error.</param>
-        /// <param name="requestTimeout">Number of seconds to wait for a request to complete before terminating the request</param>
-        public FlagsmithClient(
-            string environmentKey,
-            string apiUrl = _defaultApiUrl,
-            ILogger logger = null,
-            Func<string, Flag> defaultFlagHandler = null,
-            bool enableAnalytics = false,
-            bool enableClientSideEvaluation = false,
-            int environmentRefreshIntervalSeconds = 60,
-            Dictionary<string, string> customHeaders = null,
-            int retries = 1,
-            double? requestTimeout = null,
-            HttpClient httpClient = null)
+        public FlagsmithClient(ILogger<FlagsmithClient> logger, IFlagsmithClientConfig config, HttpClient client = null)
```

Comment:
It seems _slightly_ odd to me that the logger is the first argument to this constructor but perhaps I'm not familiar enough with C#.

Reply:
no worry here, it will be constructed by DI, all needed dependencies (constructor parameters) will be automatically constructed/reused. you can check web project (it might be in following PR)